pkg/s3mapprovider/provider.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package s3provider
```

Comment:
```suggestion
package s3mapprovider
```

Typically the package name matches the name of the last part of the import path.

This needs its own `go.mod` and an entry added to `versions.yaml` at the root of the repo.

Reply:
Fixed!